packages/app/src/embed/components/Sidebar/FileTree/index.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import * as Icons from './icons';
+import { FileContainer, IconContainer, FileName } from './elements';
+import {
+  sortingFunction,
+  isRootLevel,
+  getDepth,
+  isChildSelected,
+} from './utils';
+
+function FileTree({ sandbox, currentModuleId, setCurrentModuleId }) {
+  const { modules, directories } = sandbox;
+
+  const allFiles = [...directories, ...modules].map(file => ({
```

Comment:
🔖 this could really benefit from a refactor that converts the array into a tree before render